packages/models/src/actor/actor-reference.ts

```diff
@@ -15,10 +15,22 @@ export class ActorReference {
     return new ActorReference(name, path, protocol)
   }
 
+  static newWithPattern(module: ConstructorFunction, path = '/'): ActorReference | undefined {
+    if (module) {
+      const pattern = Reflect.getMetadata(ProviderKey.Actor, module)?.ref
+      if (pattern && pattern instanceof ActorReference) {
+        const ref = new ActorReference(pattern.name, path, pattern.protocol)
+        ref.matchParams(pattern)
+        return ref
+      }
+    }
```

Comment:
Retrun early to avoid nested `if` statements

Reply:
Why instantiate `ref` again when `pattern` is an instance of `ActorReference`

Reply:
> Why instantiate `ref` again when `pattern` is an instance of `ActorReference`

I use the `pattern` to save the `router` in which the arguments are defined, like `local://a/b/:arg/c`.
And this method is to initiate a `ref` with `name` and `protocol` which are from pattern. Besides, the `path` with parameters is provided by the caller, and the parameters are parsed with `pattern`.

So the `pattern` is like a definition, and the `ref` is like an implementation.